functors task (1)/student/include/impl.h

```diff
@@ -40,9 +56,16 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink alcohol, NonAlcoholDrink non_alcohol)
     {
+        static const std::map<std::pair<AlcoholDrink, NonAlcoholDrink>, Cocktail> recipes = {
```

Comment:
Beautiful solution :)
Mapping is the best choice here.
Instead of a long switch, you should prefer value mapper.